.github/workflows/docker-image.yml

```
@@ -0,0 +1,115 @@
+name: docker image building and pushing
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches: [ '*' ]
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    # Will build everyday at 4:30 and 16:30
+    - cron:  '30 4,16 * * *'
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:  
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+        with:
+            fetch-depth: 0
+      -
+        name: Retrieve modified files
+        id: changes
+        run: echo "::set-output name=files::$(git diff --name-only ${{ github.event.before }} ${{ github.sha }})"
+      -
+        name: Check for Dockerfile and context changes
+        run: |
+          if [[ $(echo "${{ steps.changes.outputs.files }}" | grep -E "(^|/)Dockerfile($|/)|(^|/)context/") ]]; then
+            echo "Dockerfile or context directory has changed."
+            echo Changed files are 
+            echo "${{ steps.changes.outputs.files }}"
+          else
+            echo "No Dockerfile or context directory changes. Skipping Docker image build and push steps."
+            echo changed files are 
+            echo "${{ steps.changes.outputs.files }}"   
+            exit 0
+          fi
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      -
+        name: Login to DockerHub
+        if: contains(steps.changes.outputs.files, 'Dockerfile')
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Extract branch name
+        if: contains(steps.changes.outputs.files, 'Dockerfile')
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/} | sed -e 's#/#-#g')"
+        id: extract_branch
+      -
+        name: Build and push a simple bundled instance
+        if: contains(steps.changes.outputs.files, '00_old_one_from_proposal/Dockerfile')
+        uses: docker/build-push-action@v4
+        with:
+          context: ./00_old_one_from_proposal
+          platforms: linux/amd64, linux/aarch64
+          push: true
+          tags: ash191245141/jenkinsci-tutorials:00_old_one_${{ steps.extract_branch.outputs.branch }}
```

please use env.

Example:
```suggestion
          tags: ${{ env.DOCKERHUB_USERNAME }}/jenkinsci-tutorials:00_old_one_${{ steps.extract_branch.outputs.branch }}
```